pandas/core/generic.py

```diff
@@ -663,21 +665,36 @@ def droplevel(self: FrameOrSeries, level, axis=0) -> FrameOrSeries:
         result = self.set_axis(new_labels, axis=axis, inplace=False)
         return result
 
-    def pop(self: FrameOrSeries, item) -> FrameOrSeries:
-        """
-        Return item and drop from frame. Raise KeyError if not found.
+    _shared_docs[
+        "pop"
+    ] = """
+        Return item and drops from %(klass)s. Raise KeyError if not found.
 
         Parameters
         ----------
-        item : str
-            Label of column to be popped.
+        item : %(axes_single_arg)s
+            %(pop_item_param_info)s
```

Comment:
Are you sure `axes_single_args` is correct here? What happens if you call `.pop('index')`?

Reply:
> Thanks for attaching the screenshots. So, now, the DataFrame.pop docstring have say "DataFrame" instead of "DataFrame/Series".
> 
> > I don't know how to check code in my local machine
> 
> Have you read through the [contributing guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html)?

Got it. Checked in my Local env. and it works perfectly.

Reply:
removed it. Now it doesn't need any variable from `_shared_doc_kwargs`.